File: An II/Sem I/MAP/Lab/Practic_Mutu_Vlad/src/main/java/org/example/practic_mutu_vlad/Domain/Entity.java
package org.example.practic_mutu_vlad.Domain;

public class Entity<ID> {
    private ID id;

    public ID getId() {
        return id;
    }

    public void setId(ID id) {
        this.id = id;
    }

    @Override
    public String toString() {
        return "Entity{" +
                "id=" + id +
                '}';
    }

    @Override
    public boolean equals(Object obj) {
        if (this == obj) {
            return true;
        }
        if (obj == null || getClass() != obj.getClass()) {
            return false;
        }
        Entity<?> entity = (Entity<?>) obj;
        return id.equals(entity.id);
    }

    @Override
    public int hashCode() {
        return id.hashCode();
    }
}

File: An II/Sem I/MAP/Lab/Lab10/Domain/Entity.cs
namespace Lab10.Domain
{
    internal class Entity<ID>
    {
        public Entity() { }
        public Entity(ID id) 
        { 
            this.id = id;   
        }

        private ID id;
        
        public ID Id 
        { 
            get
            {
                return id;
            }
            set
            {
                id = value;
            } 
        }
    }
}